parquet-common/src/main/java/org/apache/parquet/io/SeekableInputStream.java

```diff
@@ -23,6 +23,10 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.function.IntFunction;
+
+import org.apache.hadoop.fs.FileRange;
```

Comment:
I feel like this might be an issue.  We probably don't want to introduce a Hadoop dependency here because it breaks the separation from Hadoop in the IO path.

Reply:
Yeah, I agree with this. One solution would be to :
Introduce a custom ParquetFileRange class in the Parquet io module, use it in the interface and convert ParquetFileRange to hadoop FileRange in the implementation in H1SeekableInputStream and H2SeekableInputStream stream.

Reply:
@danielcweeks  Do you agree with my suggested approach? Thanks for reviewing the code.